VideoLocker/src/main/java/org/edx/mobile/view/CourseHandoutFragment.java

```diff
@@ -70,48 +80,47 @@ public void onActivityCreated(Bundle savedInstanceState) {
         super.onActivityCreated(savedInstanceState);
 
         if (!(NetworkUtil.isConnected(getActivity()))) {
-            showHandoutsOffline();
-        }
-
-        try {
-            final Bundle bundle = getArguments();
-            EnrolledCoursesResponse courseData = (EnrolledCoursesResponse) bundle
-                    .getSerializable(ENROLLMENT);
-            loadData(courseData);
-        } catch (Exception ex) {
-            logger.error(ex);
+            showErrorMessage(R.string.reset_no_network_message, FontAwesomeIcons.fa_wifi);
+        } else {
+            loadData();
         }
     }
 
-    private void loadData(EnrolledCoursesResponse enrollment) {
-        GetHandoutTask task = new GetHandoutTask(getActivity(), enrollment) {
+    private void loadData() {
+        GetHandoutTask task = new GetHandoutTask(getActivity(), courseData) {
 
             @Override
             public void onSuccess(HandoutModel result) {
-                try {
-                    if(result!=null&&(!TextUtils.isEmpty(result.handouts_html))){
-                        populateHandouts(result);
-                    }else{
-                        showEmptyHandoutMessage();
-                    }
-                } catch (Exception ex) {
-                    logger.error(ex);
-                    showEmptyHandoutMessage();
+                if (getActivity() == null) {
```

Comment:
It would be better if the task was actually cancelled in `onDestroyView()`, but I guess it's not a big deal.